python/pylibcudf/pylibcudf/nvtext/tokenize.pyx

```diff
@@ -0,0 +1,286 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+from cython.operator cimport dereference
+from libcpp.memory cimport unique_ptr
+from libcpp.utility cimport move
+from pylibcudf.column cimport Column
+from pylibcudf.libcudf.column.column cimport column
+from pylibcudf.libcudf.column.column_view cimport column_view
+from pylibcudf.libcudf.nvtext.tokenize cimport (
+    character_tokenize as cpp_character_tokenize,
+    count_tokens as cpp_count_tokens,
+    detokenize as cpp_detokenize,
+    load_vocabulary as cpp_load_vocabulary,
+    tokenize as cpp_tokenize,
+    tokenize_with_vocabulary as cpp_tokenize_with_vocabulary,
+)
+from pylibcudf.libcudf.scalar.scalar cimport string_scalar
+from pylibcudf.libcudf.scalar.scalar_factories cimport (
+    make_string_scalar as cpp_make_string_scalar,
+)
+from pylibcudf.libcudf.types cimport size_type
+
+
+cdef class TokenizeVocabulary:
+    """The Vocabulary object to be used with ``tokenize_with_vocabulary``.
+
+    For details, see :cpp:class:`cudf::nvtext::tokenize_vocabulary`.
+    """
+    def __cinit__(self, Column vocab):
+        cdef column_view c_vocab = vocab.view()
+        with nogil:
+            self.c_obj = move(cpp_load_vocabulary(c_vocab))
+
+cpdef Column tokenize_scalar(Column input, Scalar delimiter=None):
+    """
+    Returns a single column of strings by tokenizing the input
+    strings column using the provided characters as delimiters.
+
+    For details, see cpp:func:`cudf::nvtext::tokenize`
+
+    Parameters
+    ----------
+    input : Column
+        Strings column to tokenize
+    delimiter : Scalar
+        String scalar used to separate individual
+        strings into tokens
+
+    Returns
+    -------
+    Column
+        New strings columns of tokens
+    """
+    cdef unique_ptr[column] c_result
+
+    if delimiter is None:
+        delimiter = Scalar.from_libcudf(
+            cpp_make_string_scalar("".encode())
+        )
+
+    with nogil:
+        c_result = cpp_tokenize(
+            input.view(),
+            dereference(<const string_scalar*>delimiter.c_obj.get()),
+        )
+
+    return Column.from_libcudf(move(c_result))
+
+cpdef Column tokenize_column(Column input, Column delimiters):
+    """
+    Returns a single column of strings by tokenizing the input
+    strings column using multiple strings as delimiters.
+
+    For details, see cpp:func:`cudf::nvtext::tokenize`
+
+    Parameters
+    ----------
+    input : Column
+        Strings column to tokenize
+    delimiters : Column
+        Strings column used to separate individual strings into tokens
+
+    Returns
+    -------
+    Column
+        New strings columns of tokens
+    """
+    cdef unique_ptr[column] c_result
+
+    with nogil:
+        c_result = cpp_tokenize(
+            input.view(),
+            delimiters.view(),
+        )
+
+    return Column.from_libcudf(move(c_result))
+
+cpdef Column count_tokens_scalar(Column input, Scalar delimiter=None):
```

Comment:
For the python reviewer: I split `tokenize` into `tokenize_scalar` and `tokenize_column` because I don't think there's a good way to have fused types with default values (ie. ColumnOrScalar delimiter = None). Please correct me if I'm wrong.